class-wp-bootstrap-navwalker.php

```diff
@@ -246,8 +246,18 @@ public function start_el( &$output, $item, $depth = 0, $args = array(), $id = 0
 				$icon_html = '<i class="' . esc_attr( $icon_class_string ) . '" aria-hidden="true"></i> ';
 			}
 
-			/** This filter is documented in wp-includes/post-template.php */
-			$title = apply_filters( 'the_title', esc_html( $item->title ), $item->ID );
+			/**
+			 * Allows titles to be optionally escaped by setting 'escape_titles' => true/false in the wp_nav_menu() args.
+			 * wp_nav_menu(array( 'escape_titles'  => false, 'walker' => new WP_Bootstrap_Navwalker() ));
+			 *
+			 * Not having escape_titles set, or setting escape_titles to true will result in escaped titles.
+			 */
+			if (property_exists($args, 'escape_titles') && ! $args->escape_titles) {
```

Comment:
`! $args->escape_titles` will return `true` for `'escape_titles'  => ''` and `'escape_titles'  => 0`. Is this intended behaviour?

Reply:
If `escape_titles` is not set, the titles will remain escaped.
If `escape_titles` is `true` the titles will be escaped
if `escape_titles` is `false` the titles will not be escaped.

Reply:
Changed `false !== $args->escape_titles` to `false == $args->escape_titles`

Reply:
> Changed `false !== $args->escape_titles` to `false == $args->escape_titles`

You're right, sorry. However, you're still using loose comparison. `'escape_titles' => 0`, `'escape_titles' => '0'`, `'escape_titles' => ''` or `'escape_titles' => null` will also result in non-escaped titles which is fine, if that's what it is supposed to do. 

| Example | Result |
| --- | --- |
| false == '' | true |
| false == 0 | true |
| false == '0' | true |
| false == null | true |

I suggest using strict comparision. If you stick to loose comparison, you should tell PHPCS to ignore the loose comparison sniff otherwise it will throw a warning here. It also signals others that you did the loose comparison on purpose.

Reply:
👍

Reply:
@IanDelMar Would you be able to point me in the PHPCS direction, I haven't worked with that before. it looks like I would use `// WPCS: loose comparison ok.`?